File: compiler/src/main/java/com/bennyhuo/tieguanyin/compiler/basic/RequiredField.java
package com.bennyhuo.tieguanyin.compiler.basic;

import com.sun.tools.javac.code.Symbol;

/**
 * Created by benny on 1/29/18.
 */

public class RequiredField implements Comparable<RequiredField>{
    private static final String CONSTS_REQUIRED_FIELD_PREFIX = "REQUIRED_";

    private final Symbol.VarSymbol symbol;
    private final String name;

    public RequiredField(Symbol.VarSymbol symbol) {
        this.symbol = symbol;
        name = symbol.getQualifiedName().toString();
    }

    public Symbol.VarSymbol getSymbol() {
        return symbol;
    }

    public String getName() {
        return name;
    }

    public String getPrefix(){
        return CONSTS_REQUIRED_FIELD_PREFIX;
    }

    @Override
    public int compareTo(RequiredField requiredField) {
        return name.compareTo(requiredField.name);
    }
}

File: compiler/src/main/java/com/bennyhuo/tieguanyin/compiler/basic/RequiredField.kt
package com.bennyhuo.tieguanyin.compiler.basic

import com.bennyhuo.tieguanyin.compiler.utils.KotlinTypes
import com.squareup.javapoet.ClassName
import com.sun.tools.javac.code.Symbol

/**
 * Created by benny on 1/29/18.
 */

open class RequiredField(private val symbol: Symbol.VarSymbol) : Comparable<RequiredField> {
    val name = symbol.qualifiedName.toString()

    open val prefix = "REQUIRED_"

    val isPrivate = symbol.isPrivate

    val isPrimitive = symbol.type.isPrimitive

    fun asTypeName() = ClassName.get(symbol.type)
    fun asKotlinTypeName() = KotlinTypes.toKotlinType(symbol.type)

    override fun compareTo(requiredField: RequiredField): Int {
        return name.compareTo(requiredField.name)
    }
}
